### PR TITLE
BLE: Handle new addresses type in GenericGap.

### PR DESCRIPTION
**Pull request by pan-**

### CODE REVIEW

**Comment by pan-** [comment]
@paul-szczepanek-arm Could you review ?

File: features/FEATURE_BLE/source/generic/GenericGap.cpp [hunk 1]
```diff
@@ -314,7 +314,9 @@ static bool is_whitelist_valid(const Gap::Whitelist_t& whitelist)
 
     for (size_t i = 0; i < whitelist.size; ++i) {
         const BLEProtocol::Address_t& address = whitelist.addresses[i];
-        if (address.type > BLEProtocol::AddressType::RANDOM_PRIVATE_NON_RESOLVABLE) {
```
Comment (paul-szczepanek-arm) [inline features/FEATURE_BLE/source/generic/GenericGap.cpp]: well that is a bit evil when it comes to enums when they get new items

Comment (pan-) [reply features/FEATURE_BLE/source/generic/GenericGap.cpp]: It is!
But it's not worst than a straight cast 😆 .

**Comment by paul-szczepanek-arm** [comment]
looks right to me but is there no way to somehow normalise types, having similar yet different types is just begging for subtle hard to find errors

**Comment by pan-** [comment]
Yeah there's an easy way: breaking backward compatibility. There's no value to keep incorrect concepts but that's against mbed OS policies ...

**Pull request merged**